SCons/Tool/__init__.py

```diff
@@ -673,10 +673,14 @@ class _LibSonameGenerator(_LibInfoGeneratorBase):
     """Library soname generator. Returns library soname (e.g. libfoo.so.0) for
     a given node (e.g. /foo/bar/libfoo.so.0.1.2)"""
 
-    def __init__(self, libtype):
+    def __init__(self, libtype=None):
         super(_LibSonameGenerator, self).__init__(libtype, 'Soname')
 
-    def __call__(self, env, libnode, **kw):
+    def __call__(self, env, libnode, *args, **kw):
```

Comment:
this got pulled in from the SONAME_GENERATOR by accident, need to remove this change

Reply:
removed in 027be11